diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 6ab89553..fba64d54 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -97,6 +97,8 @@ static const I2CDeviceTypeMap sensorTypes{
     {"XDPE11280", I2CDeviceType{"xdpe11280", true}},
     {"XDPE12284", I2CDeviceType{"xdpe12284", true}},
     {"XDPE152C4", I2CDeviceType{"xdpe152c4", true}},
+    {"LTC2945", I2CDeviceType{"ltc2945", true}},
+    {"LTC4287", I2CDeviceType{"ltc4287", true}},
 };
 
 enum class DevTypes
